ui/apps/platform/src/Containers/MainPage/InviteUsersModal.tsx

```diff
@@ -1,21 +1,126 @@
+import { AuthProvider } from 'services/AuthService/AuthService';
+import { dedupeDelimtedString } from 'utils/textUtils';
+import { mergeGroupsWithAuthProviders } from '../AccessControl/AuthProviders/authProviders.utils';
+import InviteUsersForm from './InviteUsersForm';
+
+type InviteFormValues = {
+    emails: string;
+    provider: string;
+    role: string;
+};
+
+// email validation from discussion in Yup repo,
+// https://github.com/jquense/yup/issues/564#issuecomment-536068508
+const isEmail = (value) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value);
+const validationSchema = yup.object().shape({
+    emails: yup
+        .string()
+        .transform((value) => dedupeDelimtedString(value).join(',')) // dedupe - optional step
+        .required('At least one email is required')
+        .test('emails', 'Invalid email address', (value) =>
+            Boolean(
+                value &&
+                    value
+                        .split(',')
+                        .map((v) => v.trim())
+                        .every(isEmail)
+            )
+        ), // .string().trim().required('At least one email is required'),
+    provider: yup.string().required('An auth provider is required'),
+    role: yup.string().required('A role is required'),
+});
+
+const defaultInviteFormValues: InviteFormValues = {
+    emails: '',
+    provider: '',
+    role: '',
+};
 
 const feedbackState = createStructuredSelector({
+    authProviders: selectors.getAvailableAuthProviders,
+    groups: selectors.getRuleGroups,
+    roles: selectors.getRoles,
+    isFetchingAuthProviders: (state) =>
+        selectors.getLoadingStatus(state, authActionTypes.FETCH_AUTH_PROVIDERS) as boolean,
+    isFetchingRoles: (state) =>
+        selectors.getLoadingStatus(state, roleActionTypes.FETCH_ROLES) as boolean,
     invite: selectors.inviteSelector,
 });
 
 function InviteUsersModal(): ReactElement | null {
     const { invite: showInviteModal } = useSelector(feedbackState);
     const dispatch = useDispatch();
+    const formik = useFormik<InviteFormValues>({
+        initialValues: defaultInviteFormValues,
+        onSubmit: () => {}, // required but not used, because the submit action is in the modal footer
+        validationSchema,
+        validateOnMount: true,
+    });
+
+    const { isValid, values, setFieldValue } = formik;
+
+    const { authProviders, groups, roles } = useSelector(feedbackState);
+
+    const authProvidersWithRules = mergeGroupsWithAuthProviders(authProviders, groups);
+
+    const allowedRoles = roles.filter((role) => {
+        return role.name !== 'Admin' && role.name !== 'None';
+    });
+
+    useEffect(() => {
+        dispatch(authActions.fetchAuthProviders.request());
+        dispatch(roleActions.fetchRoles.request());
+        dispatch(groupActions.fetchGroups.request());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const typedProviders = authProviders as AuthProvider[]; // redux state from sagas is not yet typed
```

Comment:
Is it worthwhile to add something like the following here until redux state is typed to avoid runtime errors?


```suggestion
        if (!Array.isArray(authProviders)) {
            return;
        }
        const typedProviders = (authProviders as AuthProvider[]).filter((provider) => 
            (typeof provider.name === 'string')
        ); // redux state from sagas is not yet typed
```

Reply:
The default Redux state for the list of auth providers is an empty array, but you make a good point--I can see a bad API response corrupting the Redux state, so I'll implement that that. Worthwhile even if Redux state is typed.